fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -661,6 +678,113 @@ where
 
         Ok(())
     }
+
+    /// Loads blob ID pointed by `a`, and then for that blob,
+    /// copies `c` bytes from it starting from offset `b` into the stack.
+    /// ```txt
+    /// contract_id = mem[$rA, 32]
+    /// contract_code = contracts[contract_id]
+    /// mem[$ssp, $rC] = contract_code[$rB, $rC]
+    /// ```
+    /// Returns the total length of the contract code that was loaded from storage.
+    pub(crate) fn load_blob_code(
+        mut self,
+        blob_id_addr: Word,
+        blob_offset: Word,
+        length_unpadded: Word,
+    ) -> IoResult<(), S::DataError>
+    where
+        S: InterpreterStorage,
+    {
+        let ssp = *self.ssp;
+        let sp = *self.sp;
+        let region_start = ssp;
+
+        if ssp != sp {
+            return Err(PanicReason::ExpectedUnallocatedStack.into())
+        }
+
+        let blob_id = BlobId::from(self.memory.read_bytes(blob_id_addr)?);
+        let blob_offset: usize = blob_offset
+            .try_into()
+            .map_err(|_| PanicReason::MemoryOverflow)?;
+
+        let current_contract = current_contract(self.context, self.fp, self.memory)?;
+
+        let length = bytes::padded_len_usize(
+            length_unpadded
+                .try_into()
+                .map_err(|_| PanicReason::MemoryOverflow)?,
+        )
+        .map(|len| len as Word)
+        .unwrap_or(Word::MAX);
```

Comment:
```suggestion
        let length = bytes::padded_len_word(length_unpadded).unwrap_or(Word::MAX);
```

It produces different results on the 32-bit and 64-bit platforms.